### ISSUE TITLE
Add API route for user bid history

### ISSUE THREAD

**Issue by sbddesign** [PlebFM/PlebFM#18]
We need an API route that will help us fetch all songs a user has bid on. This will help me finish up #9, which is the user profile screen and shows all bid history.

**Comment by bnonni** [comment]
Claiming this issue. Working on it now.

**Comment by bnonni** [comment]
Reopening to discuss outstanding questions re: filtering bid history

Asking in relation to a broader question / data design decision we need to make (unless there's a better way to query that idk about which is totally possible).

when querying the Play collection, I'm using the userId to find the bids for that user. However, if I use .find() i get all the bids for that user, the associated data in the Play object AND the other bids that are not attached to that userId since I'm not filtering deeply into the array.
E.g. response from GET /api/bid/history?userId=clb9vyqmg0007ufyxb5049st0 using .find()
```
[{
    _id: ObjectId("6390c04a9388e587a9be7d3d"),
    songId: '5EWPGh7jbTNO2wakv8LjUI',
    status: 'queued',
    queueTimestamp: '1670430794061',
    bids: [
      {
        userId: 'clb9vyqmg0007ufyxb5049st0',
        bidAmount: 25,
        timestamp: '1672504884943',
        rHash: 'd451f5859c0b24ef1e90363510d0c02f5a3afcd61e3b46739c6f2726ef7b307f'
      },
      {
        userId: 'clb9u0o4v0000x98fflh5ftiq',
        bidAmount: 17,
        timestamp: '1672505562332',
        rHash: '8aacd7f5d2e2af163ed232bfb62aa13e0dbd98f959e5e033814c7c96caea186a'
      }
    ],
    runningTotal: 95,
    __v: 0,
    hostId: 'nonce33jfkasjkdf',
    playId: 'clccbkbgx0001uwd83xkl5c2v'
  },
  {
    _id: ObjectId("639155119dc367a1e2bc4b46"),
    songId: '01Lr5YepbgjXAWR9iOEyH1',
    status: 'queued',
    queueTimestamp: '1670468881012',
    bids: [
      {
        userId: 'clb9vyqmg0007ufyxb5049st0',
        bidAmount: 20,
        timestamp: '1670468881012',
        rHash: 'c18e7e3e5f9c0fc3f974dae001bf4fd75f63dc53660ebf29ac0ad64f1a3d164c'
      }
    ],
    runningTotal: 20,
    __v: 0,
    hostId: 'nonce33jfkasjkdf',
    playId: 'clccbkc360002uwd8033sfhjt'
  },
  {
    _id: ObjectId("63b06635c7c1be86e50aed68"),
    songId: '5uU1uuyaUIBOiiUSQG7Wl3',
    status: 'queued',
    queueTimestamp: '1672504884943',
    bids: [
      {
        userId: 'clb9w3wm40008ufyx9553dghn',
        bidAmount: 25,
        timestamp: '1672504884943',
        rHash: 'd451f5859c0b24ef1e90363510d0c02f5a3afcd61e3b46739c6f2726ef7b307f'
      },
      {
        userId: 'clb9vyqmg0007ufyxb5049st0',
        bidAmount: 17,
        timestamp: '1672505562332',
        rHash: '8aacd7f5d2e2af163ed232bfb62aa13e0dbd98f959e5e033814c7c96caea186a'
      }
    ],
    runningTotal: 42,
    __v: 0,
    hostId: 'nonce33jfkasjkdf',
    playId: 'clccbkcfb0003uwd8d38w0wl5'
  }]
```
 In this design, I have to use nested .filter() methods to remove bids without that userId.
```js
const plays = await Plays.find({
            bids: { $elemMatch: { userId: req.query.userId } }
        });

const bids = plays.filter((play: Play) =>
            play.bids.filter((bid: any) => {
                return bid.userId === userId;
            })
        );

return res.status(200).json({ success: true, new: false, message: bids });
```

On the flip, if I use .aggregate, I can filter out the bids in each Play that do have the userId and return only those leaving out extra bids on that play by other users. However, I ONLY get back an array of bids and no Play info.
E.g. response from GET /api/bid/history?userId=clb9vyqmg0007ufyxb5049st0 using .aggregate()
```
[{
            "_id": "6390c04a9388e587a9be7d3d",
            "bids": [
                {
                    "userId": "clb9vyqmg0007ufyxb5049st0",
                    "bidAmount": 25,
                    "timestamp": "1672504884943",
                    "rHash": "d451f5859c0b24ef1e90363510d0c02f5a3afcd61e3b46739c6f2726ef7b307f"
                }
            ]
        },
        {
            "_id": "639155119dc367a1e2bc4b46",
            "bids": [
                {
                    "userId": "clb9vyqmg0007ufyxb5049st0",
                    "bidAmount": 20,
                    "timestamp": "1670468881012",
                    "rHash": "c18e7e3e5f9c0fc3f974dae001bf4fd75f63dc53660ebf29ac0ad64f1a3d164c"
                }
            ]
        },
        {
            "_id": "63b06635c7c1be86e50aed68",
            "bids": [
                {
                    "userId": "clb9vyqmg0007ufyxb5049st0",
                    "bidAmount": 17,
                    "timestamp": "1672505562332",
                    "rHash": "8aacd7f5d2e2af163ed232bfb62aa13e0dbd98f959e5e033814c7c96caea186a"
                }
            ]
        }]
```

**Comment by bnonni** [comment]
- Use first design
- User is first in list
- Need User object, not userId

**Comment by sbddesign** [comment]
Closed with #64 
Yay @bnonni 🎉
